internal/command/meta_backend.go

```diff
@@ -944,6 +944,42 @@ func (m *Meta) backendFromConfig(opts *BackendOpts) (backend.Backend, tfdiags.Di
 		// > Changing how the store is configured.
 		// > Allowing values to be moved between partial overrides and config
 
+		// We are not going to migrate if...
+		//
+		// The state storage provider is the same
+		// AND the provider version is the same
+		// AND the provider config cache hash values match, indicating that the provider config is valid and completely unchanged.
+		// AND the same state_store implementation is used
+		// AND the state_store config cache hash values match, indicating that the state_store config is valid and completely unchanged.
+		// AND we're not providing any overrides. An override can mean a change overriding an unchanged state_store block (indicated by the hash value).
+		pVersion, vDiags := getStateStorageProviderVersion(stateStoreConfig, opts.Locks)
+		diags = diags.Append(vDiags)
+		if vDiags.HasErrors() {
+			return nil, diags
+		}
+
+		if s.StateStore.Provider.Source.Equals(stateStoreConfig.ProviderAddr) &&
+			s.StateStore.Provider.Version.Equal(pVersion) &&
+			(uint64(stateStoreProviderHash) == s.StateStore.Provider.Hash) &&
+			s.StateStore.Type == stateStoreConfig.Type &&
+			(uint64(stateStoreHash) == s.StateStore.Hash) &&
+			(!opts.Init || opts.ConfigOverride == nil) {
```

Comment:
This is pretty dense, though is carrying-forward how the equivalent block is handled for backends.

I imagine that any refactoring here would inform how the rest of this case block is implemented. Would it be ok to save that for a subsequent PR(s)?